src/utxo/mod.rs

```diff
@@ -0,0 +1,77 @@
+use crate::bitcoin::{Block, TxOut};
+
+mod mem;
+
+#[cfg(feature = "db")]
+mod db;
+
+pub use mem::MemUtxo;
+
+use bitcoin::OutPoint;
+#[cfg(feature = "db")]
+pub use db::DbUtxo;
+
+pub trait Utxo {
+    /// Add all the outputs of all the transaction in the block in the Utxo set
+    fn add(&mut self, block: &Block, height: u32);
+
+    /// Get all the prevouts in the block at `height` in the order they are found in the block.
+    /// first element in the vector is the prevout of the first input of the first transaction after
+    /// the coinbase
+    fn get(&mut self, height: u32) -> Vec<TxOut>;
+
+    /// return stats about the Utxo
+    fn stat(&self) -> String;
```

Comment:
```
trait Utxo {
  type Stats : Display /* + serde::Serialize etc. */;

  ...
  fn stat(&self) -> Self::Stats;
}
```

Reply:
Oh, but that will not make it object-safe ... which I guess you don't use right now, so maybe OK...